### PR TITLE
[GlobalISel][AArch64] Combine unmerge(G_EXT v, undef) to unmerge(v).

### PR DESCRIPTION
**Pull request by dzhidzhoev**
When having `<N x t> d1, unused = unmerge(G_EXT <2*N x t> v1, undef, N)`, it is possible to express it just as `unused, d1 = unmerge v1`.

It is useful for tackling regressions in arm64-vcvt_f.ll, introduced in https://reviews.llvm.org/D144670.

### CODE REVIEW

**Comment by arsenm** [comment]
Should be a generic combine. Can you generalize to N unused defs?

**Comment by dzhidzhoev** [comment]
> Should be a generic combine. Can you generalize to N unused defs?

G_EXT is an opcode for AArch64's `ext` instruction. Could you please suggest how to generalize it for other targets?

**Review by aemerson — CHANGES_REQUESTED** [review]
I think this needs some MIR tests.

File: llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp [hunk 1]
```diff
@@ -1071,18 +1071,18 @@ void applyVectorSextInReg(MachineInstr &MI, MachineRegisterInfo &MRI,
 bool matchUnmergeExtToUnmerge(MachineInstr &MI, MachineRegisterInfo &MRI,
                               Register &MatchInfo) {
   assert(MI.getOpcode() == TargetOpcode::G_UNMERGE_VALUES);
-  if (MI.getNumDefs() != 2)
+  auto &Unmerge = cast<GUnmerge>(MI);
+  if (Unmerge.getNumDefs() != 2)
     return false;
-  if (!MRI.use_nodbg_empty(MI.getOperand(1).getReg()))
+  if (!MRI.use_nodbg_empty(Unmerge.getOperand(1).getReg()))
```
Comment (aemerson) [inline llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:1077]: `GenericMachineInstr` also gives you `getReg(N)`

File: llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp [hunk 2]
```diff
@@ -1071,18 +1071,18 @@ void applyVectorSextInReg(MachineInstr &MI, MachineRegisterInfo &MRI,
 bool matchUnmergeExtToUnmerge(MachineInstr &MI, MachineRegisterInfo &MRI,
                               Register &MatchInfo) {
   assert(MI.getOpcode() == TargetOpcode::G_UNMERGE_VALUES);
```
Comment (aemerson) [inline llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:1073]: This assert becomes unnecessary once you have the `cast<GUnmerge>` below.

**Review by aemerson — APPROVED** [review]
LGTM with minor changes. We can actually rename the G_EXT and friends to specify AArch64 as well in another change.

**Pull request merged**